### PR TITLE
fix: subtitle CEA708 doesn't show up cause it gets index -1

### PR DESCRIPTION
**Pull request by Yuvalke**
### Description of the Changes

set the index for the native track without language

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/track/text-track.js [hunk 1]
```diff
@@ -40,6 +46,15 @@ const TextTrack: TextTrack = class TextTrack extends Track {
     return this._external;
   }
 
+  /**
+   * Getter for the CEA of the text track.
+   * @public
+   * @returns {boolean} - The kind of the text track.
```
Comment (RoyBregman) [inline src/track/text-track.js:52]: It is copied pasted wrong :)
```suggestion
   * @returns {boolean} - returns true if the text track has no language specified
```

File: src/player.js [hunk 2]
```diff
@@ -2074,7 +2074,11 @@ export default class Player extends FakeEventTarget {
         const videoElement = this.getVideoElement();
         return videoElement ? Array.from(videoElement.textTracks).findIndex(track => (track ? track.language === textTrack.language : false)) : -1;
       };
-      this._getTextTracks().forEach(track => (track.index = getNativeLanguageTrackIndex(track)));
+      this._getTextTracks().forEach(track => {
+        if (!track.emptyLanguage) {
```
Comment (RoyBregman) [inline src/player.js:2078]: Why do we need this new property called emptyLanguage and the overhead of handing it (all the changes in the TextTrack class). Can't we just check here "if (track.language) {" ?

Comment (RoyBregman) [reply src/player.js:2078]: Oh, you want it targeted only for the NativeAdapter?

Comment (RoyBregman) [reply src/player.js:2078]: But I guess this code is correct always... Even if it doesn't happen in other adapters. If there is no language then don't try to look it up.

Comment (Yuvalke) [reply src/player.js:2078]: no cause it's not empty

Comment (Yuvalke) [reply src/player.js:2078]: we set value on native adapter for CEA708

File: src/engines/html5/media-source/adapters/native-adapter.js [hunk 3]
```diff
@@ -666,6 +666,7 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
         } else if (settings.kind === 'captions' && this._config.enableCEA708Captions) {
           settings.label = settings.label || captionsTextTrackLabels.shift();
           settings.language = settings.language || captionsTextTrackLanguageCodes.shift();
+          settings.emptyLanguage = !!settings.language;
```
Comment (RoyBregman) [inline src/engines/html5/media-source/adapters/native-adapter.js:669]: Maybe instead we can just update the textTracks on the videoElement, like this:
`            if (!!settings.language) {
              settings.language = settings.language;
            } else {
              textTracks[i].language = settings.language = captionsTextTrackLanguageCodes.shift();
            }`
Then we don't need the new prop and it works perfect.

Comment (Yuvalke) [reply src/engines/html5/media-source/adapters/native-adapter.js:669]: you can't update the textTrack on video element it's read only

Comment (RoyBregman) [reply src/engines/html5/media-source/adapters/native-adapter.js:669]: But it worked :)

Comment (RoyBregman) [reply src/engines/html5/media-source/adapters/native-adapter.js:669]: We manipulate the mode, isn't same to manipulate this also?

Comment (Yuvalke) [reply src/engines/html5/media-source/adapters/native-adapter.js:669]: https://developer.mozilla.org/en-US/docs/Web/API/TextTrack

Comment (RoyBregman) [reply src/engines/html5/media-source/adapters/native-adapter.js:669]: Yes... You are right. Bummer. Then lets stick with your original solution. Seems to be the best (and only) option

**Review by RoyBregman — APPROVED** [review]

**Comment by RoyBregman** [comment]
fix conflicts

**Pull request closed without merging**